packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts

```diff
@@ -137,6 +159,22 @@ export default class AmazonPayV2PaymentProcessor {
         return isPh4Enabled;
     }
 
+    private _prepareRequestConfig(
+        createCheckoutSessionConfig: Required<AmazonPayV2CheckoutSessionConfig>,
+        estimatedOrderAmount?: AmazonPayV2Price,
+        productType?: AmazonPayV2PayOptions,
+    ): RequestConfig {
+        const { publicKeyId, ...signedPayload } = createCheckoutSessionConfig;
+
+        return {
+            createCheckoutSessionConfig: this._isEnvironmentSpecific(publicKeyId)
+                ? signedPayload
+                : createCheckoutSessionConfig,
+            ...(estimatedOrderAmount && { estimatedOrderAmount }),
+            ...(productType && { productType }),
```

Comment:
I'm not sure if Amazon's script accepts modifying this value. I've found nothing about it in their docs.

Reply:
@mauricio-sg we can change `productType` and Amazon allows to do this.
There is an example you can see: [existing_buyer_flow](https://amazonpay-onboarding.com/ee/flow/existing_buyer_flow/flow3/)

Reply:
Cool, I didn't know that. I have to mention that I didn't know the existence of that page and, the same page says in its introduction that _"If there are any discrepancies between description in this document and Integration Guide, the [Integration Guide](https://developer.amazon.com/docs/amazon-pay-checkout/introduction.html) shall prevail."_ which is the one that Amazon provided us with to work. When I said, _"I've found nothing about it in their docs."_ I was referring to that guide, sorry for not previously posting the link. So, the button is working as we can see in the videos but it might be worth double-checking with the Amazon team if it's really fully supported.

Reply:
Hey @bc-nick can you check the comment Mauricio posted as well and provide testing screenshot for the same as well. Thanks.